src/main/scala/com/cognite/sdk/scala/v1/dataPoints.scala

```diff
@@ -4,82 +4,110 @@
 package com.cognite.sdk.scala.v1
 
 import com.cognite.sdk.scala.common.{AggregateDataPoint, DataPoint, StringDataPoint}
+import io.circe.syntax.EncoderOps
+import io.circe.{Encoder, Json}
 
 final case class DataPointsByExternalId(
     externalId: String,
     datapoints: Seq[DataPoint]
 )
 
+trait DataPointsResponse[D] {
+  def id: Long
+  def getExternalId: Option[String]
+  def isString: Boolean
+  def unit: Option[String]
+  def datapoints: Seq[D]
+}
+
 final case class DataPointsByIdResponse(
     id: Long,
     externalId: Option[String],
     isString: Boolean,
-    isStep: Boolean,
+    isStep: Boolean = true, // default value for accidental string points read
```

Comment:
What does `isStep` have to do with string points?

Reply:
the `isStep` field is not present in datapoints, so the parser crashed when you tried to read string datapoint response into this DataPointsByIdResponse object. I added this default value, so it at least does not crash in the parser.